src/com/xilinx/rapidwright/design/blocks/PBlockGenerator.java

```diff
@@ -974,38 +979,44 @@ public static void main_testing(String[] args) {
 				if(numSLICEColumns > 0 || numSLICEMColumns > 0){
 					HashMap<Integer, Integer []> CLBPBlock = new HashMap<Integer, Integer []> ();
 					createAllPBlocks (CLBPBlock,patMap,p, numSLICEColumns, numSLICEMColumns,numBRAMColumns,numDSPColumns,numSLICERows);		// Re-generate pblock to write it in the file
-					// clb_pblock value:  Integer[] {x_l,x_r,y_d,y_u}
-					int LeftX   = CLBPBlock.get(0)[0];
-					int RightX  = CLBPBlock.get(0)[1];
-					int UpperY  = CLBPBlock.get(0)[3];
-					int LowerY  = CLBPBlock.get(0)[2];
-
-					sb.append("SLICE_X" + LeftX + "Y" + LowerY + ":SLICE_X" + RightX + "Y" + UpperY);
-					
-					// Write PBlock in Global PBlock file, so that the next IPs will try to use other columns if free
-					// Current solution works actually if only 1 pblock is used for the implementation of the IP. If more are used, this won't give an accurate value
-					// Also, current solution computes free resources only in case of CLB pblocks. The BRAMS and DRAMs columns are selected to match the most suitable clb pblock
-					List<String> WritePBlocks = new ArrayList<String>();
-					for (Integer i: CLBPBlock.keySet()) {
-						WritePBlocks.add("SLICE_X" + CLBPBlock.get(i)[0] + "Y" + CLBPBlock.get(i)[2] + ":SLICE_X" + CLBPBlock.get(i)[1] + "Y" + CLBPBlock.get(i)[3]);
-					} 
-					if(IP_NR_INSTANCES==0) {
-						if (debug)
-							System.out.println(" CRITICAL WARNING: IP_NR_INSTANCES is 0! Default value 1 was set.");
-						IP_NR_INSTANCES = 1;
+					if(CLBPBlock.size()>0) { // If no feasible CLB PBlock found for this pattern, skip it!
```

Comment:
Is there a difference here?  If it is just whitespace difference, do you mind reverting this block to the original code?

Reply:
I added an if-clause (line 994 + 1025-1028) to avoid error if no feasible* PBlocks are found for a pattern. Thus, the indentation is different and probably that's why github signals this as different code.
* - probability is really low. However, I had one design where I got error when reading CLBPBlock.get(0)[0];. The createAllPBlocks() function checks if PBlock corners.getSite return null. When I got the error, the corresponding pattern had only 1 possible PBlock and one of the corners were not a valid Site.  As said, probability is quite low in my opinion, but I introduced the additional if-clauses as safety net.